YAFC/Program.cs

```diff
@@ -27,42 +27,19 @@ public static class Program {
             Font.subheader = new Font(regular, 1.5f);
             Font.text = new Font(regular, 1f);
 
-            ProjectDefinition cliProject = null;
+            ProjectDefinition cliProject = CommandLineParser.Parse(args);
 
-            if (args.Length > 0) {
-                if (args.Length < 3) {
-                    Console.WriteLine("Usage: YAFC <projectPath> <dataPath> <modsPath> [expensive]");
-                    Console.WriteLine("<projectPath> - path to the project file");
-                    Console.WriteLine("<dataPath>    - path to the data folder, e.g. C:\\Factorio\\data or /home/user/Factorio/data");
-                    Console.WriteLine("<modsPath>    - path to the mods folder, e.g. C:\\Factorio\\mods or /home/user/Factorio/mods");
-                    Console.WriteLine("expensive     - optional, if provided YAFC will use expensive recipes");
-                    return;
-                }
-
-                cliProject = new ProjectDefinition {
-                    path = args[0],
-                    dataPath = args[1],
-                    modsPath = args[2]
-                };
-
-                if (args.Length >= 4) {
-                    cliProject.expensive = args[3] == "expensive";
-                }
+            if (CommandLineParser.errorOccured || CommandLineParser.helpRequested) {
+                Console.WriteLine("YAFC CE v" + YafcLib.version.ToString(3));
+                Console.WriteLine();
 
-                if (!File.Exists(cliProject.path)) {
-                    Console.WriteLine("Project file not found: " + cliProject.path);
-                    return;
+                if (CommandLineParser.errorOccured) {
+                    Console.WriteLine($"Error: {CommandLineParser.lastError}");
+                    Console.WriteLine();
                 }
 
-                if (!Directory.Exists(cliProject.dataPath)) {
-                    Console.WriteLine("Data folder not found: " + cliProject.dataPath);
-                    return;
-                }
-
-                if (!Directory.Exists(cliProject.modsPath)) {
-                    Console.WriteLine("Mods folder not found: " + cliProject.modsPath);
-                    return;
-                }
+                CommandLineParser.PrintHelp();
+                return;
```

Comment:
It would be nice to return with a non-zero exit code when there is an error. (when people make scripts to start YAFC the script can detect failures)

Reply:
Now it returns the ubiquituous **1** :-)